### ISSUE TITLE
MD025 does not trigger when first lines are empty and/or comments

### ISSUE THREAD

**Issue by xefiry** [DavidAnson/markdownlint#1420]
The error MD025 "Multiple top-level headings in the same document" should trigger when there is more than one heading level 1 in the document.

However, if there are empty and/or comment lines before the first heading, the error does not trigger.

Example with an empty line and a comment

```markdown

<!-- omit in toc -->
# Title

- [Foo](#foo)

# Foo

```

In this example, the comment is to [ignore Title in the generation of the table of content by the extension Markdown All in One](https://github.com/yzhang-gh/vscode-markdown?tab=readme-ov-file#table-of-contents), but any comment will do.

The error can be reproduced with any amount of comment lines or empty lines before the first heading.
